src/Sentry/Laravel/Tracing/Integrations/LighthouseIntegration.php

```diff
@@ -0,0 +1,174 @@
+        $this->requestSpan = $this->previousSpan->startChild($context);
+
+        SentrySdk::getCurrentHub()->setSpan($this->requestSpan);
+    }
+
+    public function handleStartExecution(StartExecution $startExecution): void
+    {
+        if ($this->requestSpan === null) {
+            return;
+        }
+
+        /** @var \GraphQL\Language\AST\OperationDefinitionNode|null $operationDefinition */
+        $operationDefinition = $startExecution->query->definitions[0] ?? null;
+
+        if ($operationDefinition === null) {
+            return;
+        }
+
+        $this->operations[] = [$startExecution->operationName, $operationDefinition];
+
+        $context = new SpanContext;
+        $context->setOp(
+            sprintf(
+                'graphql.%s{%s}',
+                $operationDefinition->operation,
+                $startExecution->operationName ?? implode(',', $this->extractOperationNames($operationDefinition))
+            )
+        );
+
+        $this->operationSpan = $this->requestSpan->startChild($context);
+
+        SentrySdk::getCurrentHub()->setSpan($this->operationSpan);
+    }
+
+    public function handleEndExecution(EndExecution $endExecution): void
+    {
+        if ($this->operationSpan === null) {
+            return;
+        }
+
+        $this->operationSpan->finish();
+        $this->operationSpan = null;
+
+        SentrySdk::getCurrentHub()->setSpan($this->requestSpan);
+    }
+
+    public function handleEndRequest(EndRequest $endRequest): void
+    {
+        if ($this->requestSpan === null) {
+            return;
+        }
+
+        $this->requestSpan->finish();
+        $this->requestSpan = null;
+
+        SentrySdk::getCurrentHub()->setSpan($this->previousSpan);
+        $this->previousSpan = null;
+
+        $this->updateTransaction();
+
+        $this->operations = [];
+    }
+
+    private function updateTransaction(): void
+    {
+        $groupedOperations = [];
+
+        foreach ($this->operations as [$operationName, $operation]) {
+            if (!isset($groupedOperations[$operation->operation])) {
+                $groupedOperations[$operation->operation] = [];
+            }
+
+            if ($operationName === null) {
+                $groupedOperations[$operation->operation] = array_merge(
+                    $groupedOperations[$operation->operation],
+                    $this->extractOperationNames($operation)
+                );
+            } else {
+                $groupedOperations[$operation->operation][] = $operationName;
+            }
+        }
+
+        array_walk($groupedOperations, static function (array &$operations, string $operationType) {
+            sort($operations, SORT_STRING);
+
+            $operations = "{$operationType}{" . implode(',', $operations) . '}';
+        });
+
+        ksort($groupedOperations, SORT_STRING);
+
+        $transactionName = 'lighthouse?' . implode('&', $groupedOperations);
+
+        Integration::setTransaction($transactionName);
```

Comment:
So this replaces the HTTP transaction?

Reply:
Yes. The reason is not too stupid but this is not working as intended at the moment. 

Traces and actual errors are matched based on the transaction name. But since we only know the new transaction name after errors happened this is not doing what I want it to do at the moment although I also don't think this is achievable, needs a little more research to see what the drawbacks of non-matching transaction names are like they would be in this case.

Reply:
What is stopping us from setting the transaction name right when the request starts?

Reply:
`StartRequest` does not contain the operations we would need to do that and `StartExecution` is for a single execution so a batched request is not considered there yet. At this moment I believe we only have all operations at the end of the last `EndExecution` which we don't know it's the last one so we can only be sure we seen it all after `EndRequest`. But maybe I'm missing something?

Reply:
Got it, makes sense that we only have all the information after parsing all requests.

Reply:
Me too, I don't think that is the right idea either. I will not touch the request transaction name for now and just change the transaction transaction name (yes I know confusing 😉).

Reply:
To ask in terms I understand: when clicking the *Performance* tab in Sentry, will I see more then a single transaction `graphql`?

![image](https://user-images.githubusercontent.com/12158000/118801226-123a7b00-b8a1-11eb-80e1-2896e07a742c.png)

Reply:
Yes.

When you click into the transaction there is a `Related Issues` block. That only works if the transaction name (in your screenshot `graphql`) matches up between the transaction and the error.

Because we modify the transaction name (for the transactions not the errors/logs anymore) we break that link, it results in multiple transactions (grouped on transaction name) which is 100% what we want but related errors will be empty unless we figure out a good way to know the transaction name earlier so error/log events will have the correct transaction name set.

Does this make sense?

Reply:
I guess `StartOperationOrOperations` is starting to look more and more juicy 😉

Reply:
I must admit it doesn't sound too bad anymore 😅

Been thinking on other ways. But there is no other way to do it unless we independently parse the GraphQL request on StartRequest which is a terrible idea. 

I'll leave it up to you to decide on it 😬

Reply:
So... after thinking a bit more about this I settled on: f9eebd3be86a0158dc9ad87623c54fa26a5bf23b.

This might cause incorrect transaction names on batch requests but for all other instances this should work.

No need for `StartOperationOrOperations` unless batch request transaction names also need to be correct, so it would still be an improvement to have something like that.

Reply:
https://github.com/nuwave/lighthouse/releases/tag/v5.11.0 is ready.